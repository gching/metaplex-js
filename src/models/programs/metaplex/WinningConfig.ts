import { BorshSchema } from '../../utils/types';
import { WinningConfigItem } from './WinningConfigItem';

export class WinningConfig {
  static SCHEMA: BorshSchema = new Map([
    [
      WinningConfig,
      {
        kind: 'struct',
        fields: [['items', [WinningConfigItem]]],
      },
    ],
    ...WinningConfigItem.SCHEMA,
  ]);

  items: WinningConfigItem[] = [];

  constructor(args?: WinningConfig) {
    Object.assign(this, args);
  }
}
