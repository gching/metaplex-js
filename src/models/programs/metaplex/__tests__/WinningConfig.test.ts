import { WinningConfig } from '../WinningConfig';

describe('WinningConfig', () => {
  it('should instantiate with the correct values', () => {
    const inst = new WinningConfig({
      items: [],
    });

    expect(inst).toMatchSnapshot();
  });
});
