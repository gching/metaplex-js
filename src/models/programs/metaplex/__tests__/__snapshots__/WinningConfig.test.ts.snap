// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WinningConfig should instantiate with the correct values 1`] = `
WinningConfig {
  "items": Array [],
}
`;
